libxrdp/xrdp_sec.c

```diff
@@ -2298,159 +2298,190 @@ xrdp_sec_process_mcs_data_channels(struct xrdp_sec *self, struct stream *s)
+                "left %d, top %d, right %d, bottom %d, flags 0x%8.8x",
+                monitor_index,
+                monitor_layout->left,
+                monitor_layout->top,
+                monitor_layout->right,
+                monitor_layout->bottom,
+                monitor_layout->is_primary);
         }
 
-        if (client_info->minfo[index].is_primary)
+        if (monitor_index == 0)
         {
-            got_primary = 1;
+            rect.left = monitor_layout->left;
+            rect.top = monitor_layout->top;
+            rect.right = monitor_layout->right;
+            rect.bottom = monitor_layout->bottom;
+        }
+        else
+        {
+            rect.left = MIN(monitor_layout->left, rect.left);
+            rect.top = MIN(monitor_layout->top, rect.top);
+            rect.right = MAX(rect.right, monitor_layout->right);
+            rect.bottom = MAX(rect.bottom, monitor_layout->bottom);
         }
 
-        LOG(LOG_LEVEL_DEBUG,
-            "Client monitor [%d]: left= %d, top= %d, right= %d, bottom= %d, "
-            "is_primary?= %d",
-            index,
-            client_info->minfo[index].left,
-            client_info->minfo[index].top,
-            client_info->minfo[index].right,
-            client_info->minfo[index].bottom,
-            client_info->minfo[index].is_primary);
+        if (monitor_layout->is_primary)
+        {
+            got_primary = 1;
+        }
     }
 
     if (!got_primary)
     {
         /* no primary monitor was set, choose the leftmost monitor as primary */
-        for (index = 0; index < monitorCount; index++)
+        for (int monitor_index = 0; monitor_index < NumMonitor; ++monitor_index)
         {
-            if (client_info->minfo[index].left == x1 &&
-                    client_info->minfo[index].top == y1)
+            monitor_layout = description->minfo + monitor_index;
+            if (monitor_layout->left != rect.left || monitor_layout->top != rect.top)
             {
-                client_info->minfo[index].is_primary = 1;
-                break;
+                continue;
             }
+            monitor_layout->is_primary = 1;
         }
     }
 
     /* set wm geometry */
-    if ((x2 > x1) && (y2 > y1))
+    if ((rect.right > rect.left) && (rect.bottom > rect.top))
     {
-        client_info->width = (x2 - x1) + 1;
-        client_info->height = (y2 - y1) + 1;
+        description->session_width = rect.right - rect.left;
+        description->session_height = rect.bottom - rect.top;
     }
     /* make sure virtual desktop size is ok */
-    if (client_info->width > 0x7FFE || client_info->width < 0xC8 ||
-            client_info->height > 0x7FFE || client_info->height < 0xC8)
+    if (description->session_width > 0x7FFE || description->session_width < 0xC8 ||
+            description->session_height > 0x7FFE || description->session_height < 0xC8)
     {
-        LOG(LOG_LEVEL_ERROR,
+        LOG(LOG_LEVEL_INFO,
```

Comment:
I don't understand the severity change of this message. An error seems appropriate here. Am I missing something?

Reply:
I think this was leftover from my debugging while developing the change. Reverted.